modules/core/src/main/java/org/apache/ignite/internal/processors/security/IgniteSecurityProcessor.java

```diff
@@ -114,14 +124,46 @@ public IgniteSecurityProcessor(GridKernalContext ctx, GridSecurityProcessor secP
         return new OperationSecurityContext(this, old);
     }
 
+    @Override public OperationSecurityContext withContext(UUID nodeId) {
+        return withContext(nodeId, nodeId);
+    }
+
     /** {@inheritDoc} */
-    @Override public OperationSecurityContext withContext(UUID subjId) {
+    @Override public OperationSecurityContext withContext(UUID senderNodeId, UUID subjId) {
         ClusterNode node = Optional.ofNullable(ctx.discovery().node(subjId))
             .orElseGet(() -> ctx.discovery().historicalNode(subjId));
 
-        SecurityContext res = node != null ? secCtxs.computeIfAbsent(subjId,
-            uuid -> nodeSecurityContext(marsh, U.resolveClassLoader(ctx.config()), node))
-            : secPrc.securityContext(subjId);
+        SecurityContext res = null;
+
+        if (!senderNodeId.equals(subjId)) {
+            ClusterNode senderNode = Optional.ofNullable(ctx.discovery().node(senderNodeId))
+                .orElseGet(() -> ctx.discovery().historicalNode(senderNodeId));
+
+            if (checkSenderNodeSubject && senderNode.isClient()) {
+                SecuritySubjectType type = node != null ? SecuritySubjectType.REMOTE_NODE : SecuritySubjectType.REMOTE_CLIENT;
+
+                log.warning("Switched to the 'deny all' policy because a client node tries to execute a request on behalf of another node " +
```

Comment:
I think this warning should be logged if `forcibleNodeKillEnabled` is `false`

Reply:
I stay it as it was, because the logic is changed.
Now it always tries to execute a suspecting request with deny all policy, but when the forcible node kill property is enabled, it kills the node simultaneously.